### PR TITLE
Add metadata using labels.

### PR DESCRIPTION
**Pull request by rossf7**
* Add metadata using labels.
* Add make build to pass dynamic metadata to the Dockerfile.
* Add make test for Travis CI.
* Replace ADD with COPY as its recommended.
* Replace ENTRYPOINT with CMD as its better for our use case.
* Add the Dockerfile as well as the Go binary to the image to help with debugging.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Image can be overidden with an env var.
+DOCKER_IMAGE?=microscaling/microscaling
+
+# Version tag from the code.
+VERSION=`cat VERSION`
```
Comment (lizrice) [inline Makefile]: My only concern here is whether we want to tag the Docker file as version N, if it's actually version N + some commits?

Comment (lizrice) [inline Makefile]: Maybe there could be a makefile target for building the latest tagged version (maybe cat the VERSION file, use this to retrieve the git code tagged as that version number). And another target that tags it differently if it's a development build (maybe the tag includes the latest SHA?)

Comment (rossf7) [inline Makefile]: The build target now appends the SHA for development builds. I've added a release target that tags the image as latest and pushes to DockerHub.

File: Makefile [hunk 2]
```diff
@@ -1,24 +1,49 @@
...
-DOCKER_IMAGE?=microscaling/microscaling
+DOCKER_IMAGE ?= microscaling/microscaling
 
-# Version tag from the code.
-VERSION=`cat VERSION`
+# Get the latest commit.
+GIT_COMMIT = `git rev-parse --short HEAD` 
 
-default: test
+ifeq ($(MAKECMDGOALS),release)
```
Comment (lizrice) [inline Makefile]: Pull the right version of code - or barf if we are not on it?

Comment (rossf7) [inline Makefile]: Issue #19 raised to add this to the Makefile.

**Comment by lizrice** [comment]
I'm happy with this now but you should check what I did in the Makefile

**Comment by rossf7** [comment]
👍  LGTM

**Pull request merged**